addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java

```diff
@@ -255,6 +266,36 @@ private JToggleButton getButtonHostHeader() {
         return hostHeader;
     }
 
+    private JButton getButtonLowerCaseHeaderNames() {
+        if (lowerCaseHeaderNames == null) {
+            lowerCaseHeaderNames =
+                    new JButton(ExtensionRequester.createIcon("lowercase-header-button.png"));
+            lowerCaseHeaderNames.setToolTipText(
+                    Constant.messages.getString(
+                            "requester.httpsender.button.lowerCaseHeadersName"));
+            lowerCaseHeaderNames.addActionListener(e -> lowerCaseHeaderNames());
+        }
+        return lowerCaseHeaderNames;
+    }
+
+    private void lowerCaseHeaderNames() {
+        customHttpPanelRequest.saveData();
```

Comment:
Should catch exceptions and warn/inform the user (see how it behaves when changing the method with a malformed request header).

Reply:
does it look good now ? 

I have also added this in Messages.properties "requester.httppanel.lowercaseHeaderNames.warn = Unable to lowercase the header names, failed to set the data to the message."
```

    private JButton getButtonLowerCaseHeaderNames() {
        if (lowerCaseHeaderNames == null) {
            lowerCaseHeaderNames =
                    new JButton(ExtensionRequester.createIcon("lowercase-header-button.png"));
            lowerCaseHeaderNames.setToolTipText(
                    Constant.messages.getString(
                            "requester.httpsender.button.lowerCaseHeaderNames"));
            lowerCaseHeaderNames.addActionListener(
                    e -> {
                        try {
                            customHttpPanelRequest.saveData();
                        } catch (InvalidMessageDataException er) {
                            StringBuilder warnMessage = new StringBuilder(150);
                            warnMessage.append(
                                    Constant.messages.getString(
                                            "requester.httppanel.lowercaseHeaderNames.warn"));

                            String exceptionMessage = er.getLocalizedMessage();
                            if (exceptionMessage != null && !exceptionMessage.isEmpty()) {
                                warnMessage.append('\n').append(exceptionMessage);
                            }
                            View.getSingleton().showWarningDialog(warnMessage.toString());
                            return;
                        }
                        HttpMessage msg = (HttpMessage) customHttpPanelRequest.getMessage();
                        lowerCaseHeaderNames(msg);
                        customHttpPanelRequest.updateContent();
                    });
        }
        return lowerCaseHeaderNames;
    }

    static void lowerCaseHeaderNames(HttpMessage msg) {
        HttpRequestHeader httpRequestHeader = msg.getRequestHeader();
        List<HttpHeaderField> fields = httpRequestHeader.getHeaders();
        for (HttpHeaderField field : fields) {
            httpRequestHeader.setHeader(field.getName(), null);
        }
        for (HttpHeaderField field : fields) {
            httpRequestHeader.addHeader(field.getName().toLowerCase(Locale.ROOT), field.getValue());
        }
    }
```

Reply:
I think it does.